src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -1464,7 +1464,7 @@ static DeferredErrorMessage *
 NonPushableInsertSelectSupported(Query *insertSelectQuery)
 {
 	DeferredErrorMessage *deferredError = ErrorIfOnConflictNotSupported(
-		insertSelectQuery);
+		insertSelectQuery, insertSelectQuery);
```

Comment:
we should add a test with insertSelect and on conflict to see if that causes a problem as well. If so we will need to pass the original Query here as well.